akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala

```diff
@@ -278,6 +294,24 @@ object ClusterEvent {
   /**
    * INTERNAL API
    */
+  private[cluster] def diffUnreachableTeam(oldGossip: Gossip, newGossip: Gossip, selfUniqueAddress: UniqueAddress): immutable.Seq[UnreachableDataCenter] = {
+    if (newGossip eq oldGossip) Nil
+    else {
+      val oldUnreachableNodes = oldGossip.overview.reachability.allUnreachableOrTerminated
+
+      newGossip.overview.reachability.observersGroupedByUnreachable.collect {
+        case (unreachable, observers) if !oldUnreachableNodes.contains(unreachable) &&
+          unreachable != selfUniqueAddress &&
+          newGossip.member(unreachable).dataCenter != newGossip.member(selfUniqueAddress).dataCenter &&
+          observers.exists(newGossip.member(_).dataCenter == newGossip.member(selfUniqueAddress).dataCenter) ⇒
+          UnreachableDataCenter(newGossip.member(unreachable).dataCenter)
+      }(collection.breakOut)
+    }
+  }
```

Comment:
I see. I agree it is simpler to go though dcs and check for reachability. Will refactor to that.